internal/compiler/compiler.go

```diff
@@ -180,10 +180,14 @@ func (c *Compiler) ResetCache() {
 }
 
 func (c *Compiler) getSpecialVars(t *ast.Task) (map[string]string, error) {
+	entrypoint := c.Entrypoint
+	if entrypoint == "" {
+		entrypoint = "Taskfile.yml"
```

Comment:
Actually, the default is not necessarily `Taskfile.yml`. We have a list of possible 8 file names that are detected and work by default:

https://github.com/go-task/task/blob/88b095020e14defcce4645e72f503eb5d880f78d/taskfile/taskfile.go#L19-L28

In the `taskfile` package we have code that will try to detect it. It may be a bit tricky, but in order to be reliable we need to reuse that code somehow.

Reply:
I totally agree I missed that part
What I did here, export taskfile.go in a new package in order to have access to `DefaultTaskfiles`
Then I check (as it's done on `Exists` func) if the file exist. I did not reuse the Exist function because we do not want to log smth

Reply:
I checked and it works with `Taskfile.dist.yml`